Started GET "/" for 127.0.0.1 at 2017-06-14 10:53:13 +0800
Processing by Rails::WelcomeController#index as HTML
  Rendering /usr/local/rvm/gems/ruby-2.4.1/gems/railties-5.1.1/lib/rails/templates/rails/welcome/index.html.erb
  Rendered /usr/local/rvm/gems/ruby-2.4.1/gems/railties-5.1.1/lib/rails/templates/rails/welcome/index.html.erb (3.1ms)
Completed 200 OK in 135ms (Views: 6.3ms)


Started GET "/" for 127.0.0.1 at 2017-06-14 10:53:25 +0800
Processing by Rails::WelcomeController#index as HTML
  Rendering /usr/local/rvm/gems/ruby-2.4.1/gems/railties-5.1.1/lib/rails/templates/rails/welcome/index.html.erb
  Rendered /usr/local/rvm/gems/ruby-2.4.1/gems/railties-5.1.1/lib/rails/templates/rails/welcome/index.html.erb (2.4ms)
Completed 200 OK in 5ms (Views: 4.0ms)


Started GET "/" for 127.0.0.1 at 2017-06-14 11:20:35 +0800
Processing by Rails::WelcomeController#index as HTML
  Rendering /usr/local/rvm/gems/ruby-2.4.1/gems/railties-5.1.1/lib/rails/templates/rails/welcome/index.html.erb
  Rendered /usr/local/rvm/gems/ruby-2.4.1/gems/railties-5.1.1/lib/rails/templates/rails/welcome/index.html.erb (2.8ms)
Completed 200 OK in 143ms (Views: 6.3ms)


Started GET "/say" for 127.0.0.1 at 2017-06-14 11:20:44 +0800
  
ActionController::RoutingError (No route matches [GET] "/say"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/say/hello.json" for 127.0.0.1 at 2017-06-14 11:20:53 +0800
Processing by SayController#hello as JSON
Completed 500 Internal Server Error in 1ms


  
SyntaxError (/Users/miraclewong/github/RubyOnRails/apidemo/app/models/hello.rb:7: syntax error, unexpected end-of-input, expecting keyword_end
  end
     ^):
  
app/models/hello.rb:7: syntax error, unexpected end-of-input, expecting keyword_end
app/controllers/say_controller.rb:3:in `hello'
Started GET "/say/hello.json?user=Miracle" for 127.0.0.1 at 2017-06-14 11:21:15 +0800
Processing by SayController#hello as JSON
  Parameters: {"user"=>"Miracle"}
Completed 500 Internal Server Error in 1ms


  
SyntaxError (/Users/miraclewong/github/RubyOnRails/apidemo/app/models/hello.rb:7: syntax error, unexpected end-of-input, expecting keyword_end
  end
     ^):
  
app/models/hello.rb:7: syntax error, unexpected end-of-input, expecting keyword_end
app/controllers/say_controller.rb:3:in `hello'
Started GET "/say/hello.json?user=Miracle" for 127.0.0.1 at 2017-06-14 11:22:22 +0800
Processing by SayController#hello as JSON
  Parameters: {"user"=>"Miracle"}
Completed 200 OK in 1ms (Views: 0.2ms)


Started GET "/favicon.ico" for 127.0.0.1 at 2017-06-14 11:22:22 +0800
  
ActionController::RoutingError (No route matches [GET] "/favicon.ico"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/say/hello.json" for 127.0.0.1 at 2017-06-14 11:22:30 +0800
Processing by SayController#hello as JSON
Completed 200 OK in 0ms (Views: 0.1ms)


Started GET "/say/hello.json?user=Miracle" for 127.0.0.1 at 2017-06-14 11:22:39 +0800
Processing by SayController#hello as JSON
  Parameters: {"user"=>"Miracle"}
Completed 200 OK in 0ms (Views: 0.1ms)


Started GET "/say/hello" for 127.0.0.1 at 2017-06-14 11:24:52 +0800
Processing by SayController#hello as HTML
Completed 200 OK in 0ms (Views: 0.1ms)


Started GET "/say/hello?user=Miracle" for 127.0.0.1 at 2017-06-14 11:25:09 +0800
Processing by SayController#hello as HTML
  Parameters: {"user"=>"Miracle"}
Completed 200 OK in 0ms (Views: 0.1ms)


Started GET "/say" for 127.0.0.1 at 2017-06-14 11:25:13 +0800
  
ActionController::RoutingError (No route matches [GET] "/say"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/say/hello" for 127.0.0.1 at 2017-06-14 11:25:31 +0800
Processing by SayController#hello as HTML
Completed 200 OK in 0ms (Views: 0.1ms)


Started GET "/say/hello" for 127.0.0.1 at 2017-06-14 11:27:43 +0800
Processing by SayController#hello as HTML
Completed 200 OK in 0ms (Views: 0.1ms)


Started GET "/say/hello" for 127.0.0.1 at 2017-06-14 12:23:02 +0800
Processing by SayController#hello as HTML
Completed 200 OK in 129ms (Views: 0.2ms)


Started GET "/say/hello" for 127.0.0.1 at 2017-06-14 12:23:18 +0800
Processing by SayController#hello as */*
Completed 200 OK in 0ms (Views: 0.1ms)


Started GET "/say/hello?user=Miracle" for 127.0.0.1 at 2017-06-14 13:38:52 +0800
Processing by SayController#hello as HTML
  Parameters: {"user"=>"Miracle"}
Completed 200 OK in 0ms (Views: 0.1ms)


Started GET "/say/hello?user=Miracle" for 127.0.0.1 at 2017-06-14 13:39:17 +0800
Processing by SayController#hello as */*
  Parameters: {"user"=>"Miracle"}
Completed 200 OK in 0ms (Views: 0.1ms)


Started GET "/" for 127.0.0.1 at 2017-06-14 14:17:45 +0800
Processing by Rails::WelcomeController#index as HTML
  Rendering /usr/local/rvm/gems/ruby-2.4.1/gems/railties-5.1.1/lib/rails/templates/rails/welcome/index.html.erb
  Rendered /usr/local/rvm/gems/ruby-2.4.1/gems/railties-5.1.1/lib/rails/templates/rails/welcome/index.html.erb (3.7ms)
Completed 200 OK in 148ms (Views: 8.4ms)


Started GET "/login" for 127.0.0.1 at 2017-06-14 14:17:50 +0800
  
ActionController::RoutingError (No route matches [GET] "/login"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/login" for 127.0.0.1 at 2017-06-14 14:19:10 +0800
  
ActionController::RoutingError (No route matches [GET] "/login"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/login" for 127.0.0.1 at 2017-06-14 14:19:36 +0800
  
ActionController::RoutingError (No route matches [GET] "/login"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/say/login" for 127.0.0.1 at 2017-06-14 14:20:11 +0800
Processing by SayController#login as HTML
Completed 200 OK in 114ms (Views: 0.3ms)


Started GET "/say/login" for 127.0.0.1 at 2017-06-14 14:22:24 +0800
Processing by SayController#login as HTML
Completed 200 OK in 1ms (Views: 0.1ms)


Started GET "/say/login" for 127.0.0.1 at 2017-06-14 14:22:40 +0800
Processing by SayController#login as HTML
Completed 500 Internal Server Error in 1ms


  
NoMethodError (undefined method `empty?' for #<Hello:0x007f80e4c97b80>):
  
app/controllers/say_controller.rb:12:in `login'
Started GET "/say/login" for 127.0.0.1 at 2017-06-14 14:32:58 +0800
  
ActionController::RoutingError (No route matches [GET] "/say/login"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/say/login" for 127.0.0.1 at 2017-06-14 14:33:17 +0800
  
ActionController::RoutingError (No route matches [GET] "/say/login"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/say/login" for 127.0.0.1 at 2017-06-14 14:33:33 +0800
  
ActionController::RoutingError (No route matches [GET] "/say/login"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/say/login" for 127.0.0.1 at 2017-06-14 15:18:06 +0800
  
ActionController::RoutingError (No route matches [GET] "/say/login"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/say/login" for 127.0.0.1 at 2017-06-14 15:18:07 +0800
  
ActionController::RoutingError (No route matches [GET] "/say/login"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/say/login" for 127.0.0.1 at 2017-06-14 15:18:42 +0800
  
ActionController::RoutingError (No route matches [GET] "/say/login"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/say/login" for 127.0.0.1 at 2017-06-14 15:18:56 +0800
Processing by SayController#login as HTML
Completed 200 OK in 116ms (Views: 0.2ms)


Started GET "/say/login" for 127.0.0.1 at 2017-06-14 15:56:26 +0800
Processing by SayController#login as HTML
Completed 500 Internal Server Error in 1ms


  
ArgumentError (wrong number of arguments (given 2, expected 1)):
  
app/controllers/say_controller.rb:13:in `login'
Started GET "/say/login?username=wr&password=[FILTERED]" for 127.0.0.1 at 2017-06-14 15:57:26 +0800
Processing by SayController#login as HTML
  Parameters: {"username"=>"wr", "password"=>"[FILTERED]"}
Completed 500 Internal Server Error in 0ms


  
ArgumentError (wrong number of arguments (given 2, expected 1)):
  
app/controllers/say_controller.rb:13:in `login'
Started GET "/say/login?username=wr?password=123456" for 127.0.0.1 at 2017-06-14 15:57:46 +0800
Processing by SayController#login as HTML
  Parameters: {"username"=>"wr?password=123456"}
Completed 500 Internal Server Error in 0ms


  
ArgumentError (wrong number of arguments (given 2, expected 1)):
  
app/controllers/say_controller.rb:13:in `login'
Started GET "/say/login?username=wr$password=123456" for 127.0.0.1 at 2017-06-14 15:58:02 +0800
Processing by SayController#login as HTML
  Parameters: {"username"=>"wr$password=123456"}
Completed 500 Internal Server Error in 0ms


  
ArgumentError (wrong number of arguments (given 2, expected 1)):
  
app/controllers/say_controller.rb:13:in `login'
Started GET "/say/login?username=wr&password=[FILTERED]" for 127.0.0.1 at 2017-06-14 15:58:08 +0800
Processing by SayController#login as HTML
  Parameters: {"username"=>"wr", "password"=>"[FILTERED]"}
Completed 500 Internal Server Error in 0ms


  
ArgumentError (wrong number of arguments (given 2, expected 1)):
  
app/controllers/say_controller.rb:13:in `login'
Started GET "/say/login?username=wr" for 127.0.0.1 at 2017-06-14 15:58:50 +0800
Processing by SayController#login as HTML
  Parameters: {"username"=>"wr"}
Completed 500 Internal Server Error in 3ms


  
NameError (undefined local variable or method `username' for #<User:0x007fab1160b878>
Did you mean?  usernamed):
  
app/models/user.rb:8:in `login'
app/controllers/say_controller.rb:14:in `login'
Started GET "/say/login?username=wr" for 127.0.0.1 at 2017-06-14 15:59:10 +0800
Processing by SayController#login as HTML
  Parameters: {"username"=>"wr"}
Completed 200 OK in 2ms (Views: 0.2ms)


Started GET "/say/login?username=rt" for 127.0.0.1 at 2017-06-14 15:59:23 +0800
Processing by SayController#login as HTML
  Parameters: {"username"=>"rt"}
Completed 204 No Content in 0ms


Started GET "/say/login?username=l" for 127.0.0.1 at 2017-06-14 15:59:28 +0800
Processing by SayController#login as HTML
  Parameters: {"username"=>"l"}
Completed 204 No Content in 0ms


Started GET "/say/login?username=dsdd" for 127.0.0.1 at 2017-06-14 15:59:32 +0800
Processing by SayController#login as HTML
  Parameters: {"username"=>"dsdd"}
Completed 204 No Content in 0ms


Started GET "/say/login?use" for 127.0.0.1 at 2017-06-14 15:59:39 +0800
Processing by SayController#login as HTML
  Parameters: {"use"=>nil}
Completed 200 OK in 0ms (Views: 0.1ms)


Started GET "/say/login?useddd" for 127.0.0.1 at 2017-06-14 15:59:43 +0800
Processing by SayController#login as HTML
  Parameters: {"useddd"=>nil}
Completed 200 OK in 0ms (Views: 0.1ms)


Started GET "/say/login?use=ddd" for 127.0.0.1 at 2017-06-14 15:59:47 +0800
Processing by SayController#login as HTML
  Parameters: {"use"=>"ddd"}
Completed 200 OK in 0ms (Views: 0.1ms)


Started GET "/" for 127.0.0.1 at 2017-06-15 10:11:05 +0800
Processing by Rails::WelcomeController#index as HTML
  Rendering /usr/local/rvm/gems/ruby-2.4.1/gems/railties-5.1.1/lib/rails/templates/rails/welcome/index.html.erb
  Rendered /usr/local/rvm/gems/ruby-2.4.1/gems/railties-5.1.1/lib/rails/templates/rails/welcome/index.html.erb (3.6ms)
Completed 200 OK in 185ms (Views: 7.8ms)


Started GET "/users.json" for 127.0.0.1 at 2017-06-15 10:11:11 +0800
  
ActionController::RoutingError (No route matches [GET] "/users.json"):
  
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:63:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.9.1) lib/puma/configuration.rb:224:in `call'
puma (3.9.1) lib/puma/server.rb:602:in `handle_request'
puma (3.9.1) lib/puma/server.rb:435:in `process_client'
puma (3.9.1) lib/puma/server.rb:299:in `block in run'
puma (3.9.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
Started GET "/say/hello" for 127.0.0.1 at 2017-06-15 10:11:29 +0800
Processing by SayController#hello as HTML
Completed 200 OK in 1ms (Views: 0.2ms)


  [1m[35m (0.1ms)[0m  [1m[34mSELECT sqlite_version(*)[0m
  [1m[35m (1.0ms)[0m  [1m[35mCREATE TABLE "schema_migrations" ("version" varchar NOT NULL PRIMARY KEY)[0m
  [1m[35m (0.7ms)[0m  [1m[35mCREATE TABLE "ar_internal_metadata" ("key" varchar NOT NULL PRIMARY KEY, "value" varchar, "created_at" datetime NOT NULL, "updated_at" datetime NOT NULL)[0m
  [1m[35m (0.1ms)[0m  [1m[34mSELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC[0m
Migrating to CreateUsers (20170615025056)
  [1m[35m (0.0ms)[0m  [1m[36mbegin transaction[0m
  [1m[35m (0.3ms)[0m  [1m[35mCREATE TABLE "users" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "created_at" datetime NOT NULL, "updated_at" datetime NOT NULL)[0m
  [1m[35mSQL (0.1ms)[0m  [1m[32mINSERT INTO "schema_migrations" ("version") VALUES (?)[0m  [["version", "20170615025056"]]
  [1m[35m (0.5ms)[0m  [1m[36mcommit transaction[0m
  [1m[36mActiveRecord::InternalMetadata Load (0.1ms)[0m  [1m[34mSELECT  "ar_internal_metadata".* FROM "ar_internal_metadata" WHERE "ar_internal_metadata"."key" = ? LIMIT ?[0m  [["key", "environment"], ["LIMIT", 1]]
  [1m[35m (0.0ms)[0m  [1m[36mbegin transaction[0m
  [1m[35mSQL (0.2ms)[0m  [1m[32mINSERT INTO "ar_internal_metadata" ("key", "value", "created_at", "updated_at") VALUES (?, ?, ?, ?)[0m  [["key", "environment"], ["value", "development"], ["created_at", "2017-06-15 02:52:44.351587"], ["updated_at", "2017-06-15 02:52:44.351587"]]
  [1m[35m (0.5ms)[0m  [1m[36mcommit transaction[0m
  [1m[35m (0.1ms)[0m  [1m[34mSELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC[0m
  [1m[35m (0.1ms)[0m  [1m[34mSELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC[0m
